test/integration/tests/RegistrationFlow.t.sol

```diff
@@ -53,16 +59,67 @@ contract RegistrationFlowTest is MockAVSDeployer {
     }
 
     function test_RegisterDataValidator() public {
-        BN254.G1Point memory pubKey;
         vm.startPrank(whitelister);
         assertEq(chainManager.hasRole(chainManager.DATA_VALIDATOR_ROLE(), operator), false);
         chainManager.grantRole(chainManager.DATA_VALIDATOR_ROLE(), operator);
         assertEq(chainManager.hasRole(chainManager.DATA_VALIDATOR_ROLE(), operator), true);
         vm.stopPrank();
-        _registerEOOperatorWithCoordinator(operator, uint256(1), pubKey, 1000, false);
+
+        BN254.G1Point memory pubKey = BN254.hashToG1(keccak256("seed_for_hash"));
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(1);
+
+        blsApkRegistry.setBLSPublicKey(operator, pubKey);
+        _setOperatorWeight(operator, 0, 1000);
+
+        ISignatureUtils.SignatureWithSaltAndExpiry memory signature;
+        IEOBLSApkRegistry.PubkeyRegistrationParams memory params;
+        uint96[] memory stakes = new uint96[](1);
+        stakes[0] = 1000;
+        cheats.prank(operator);
+        vm.expectEmit(true, true, true, true);
+        emit DataValidatorRegistered(operator, stakes);
+        registryCoordinator.registerOperator(quorumNumbers, params, signature);
+    }
+
+    function test_UpdateDataValidator() public {
+        vm.startPrank(whitelister);
+        assertEq(chainManager.hasRole(chainManager.DATA_VALIDATOR_ROLE(), operator), false);
+        chainManager.grantRole(chainManager.DATA_VALIDATOR_ROLE(), operator);
+        assertEq(chainManager.hasRole(chainManager.DATA_VALIDATOR_ROLE(), operator), true);
+        vm.stopPrank();
+
+        BN254.G1Point memory pubKey = BN254.hashToG1(keccak256("seed_for_hash"));
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(1);
+
+        blsApkRegistry.setBLSPublicKey(operator, pubKey);
+        uint96[] memory stakes = new uint96[](1);
+        stakes[0] = 1000;
+        _setOperatorWeight(operator, 0, stakes[0]);
+
+        ISignatureUtils.SignatureWithSaltAndExpiry memory signature;
+        IEOBLSApkRegistry.PubkeyRegistrationParams memory params;
+        vm.expectEmit(true, true, true, true);
+        emit DataValidatorRegistered(operator, stakes);
+        cheats.prank(operator);
+        registryCoordinator.registerOperator(quorumNumbers, params, signature);
```

Comment:
Lines 91-104 are exact copy from the previous function. Move the code to internal function and reuse it